src/overlays/actors/ovl_Obj_Bean/z_obj_bean.c

```diff
@@ -54,137 +117,855 @@ static ColliderCylinderInit D_80B90E54 = {
+}
+
+void ObjBean_ShakeLeavesFast(ObjBean* this) {
+    this->unk_1C0 += -1;
+    if (Rand_ZeroOne() < 0.1f) {
+        this->unk_1C4 = Rand_S16Offset(0x898, 0x3E8);
+        this->unk_1CA = Rand_S16Offset(0x2EE0, 0x1F40);
+    }
+    ObjBean_UpdateLeaves(this);
+    if ((s32)this->unk_1C0 < 0) {
+        ObjBean_SetupGrow(this);
+    }
+}
+
+void ObjBean_SetupGrow(ObjBean* this) {
+    this->transformFunc = ObjBean_Grow;
+}
+
+void ObjBean_Grow(ObjBean* this) {
+    Math_StepToS(&this->leafRotFactor, 0x33E9, 0x168);
+    this->dyna.actor.scale.y = (f32)(Math_SinS(this->leafRotFactor) * 0.17434467f);
+    this->dyna.actor.scale.x = this->dyna.actor.scale.z = Math_CosS(this->leafRotFactor) * 0.12207746f;
+    ;
+}
+
+void ObjBean_SetupFlattenLeaves(ObjBean* this) {
+    this->transformFunc = ObjBean_FlattenLeaves;
+    this->leafRotFactor = 0x33E9;
+}
+
+void ObjBean_FlattenLeaves(ObjBean* this) {
+    this->leafRotFactor -= 0x960;
+    this->dyna.actor.scale.y = (f32)(Math_SinS(this->leafRotFactor) * 0.17434467f);
+    this->dyna.actor.scale.x = this->dyna.actor.scale.z = Math_CosS(this->leafRotFactor) * 0.12207746f;
+
+    if (this->leafRotFactor < 0x18E4) {
+        ObjBean_SetupGrown(this);
+    }
+}
+
+void ObjBean_SetupGrown(ObjBean* this) {
+    this->transformFunc = ObjBean_Grown;
+    this->unk_1C2 = 0xBB8;
+    this->unk_1C4 = 0;
+    this->unk_1C6 = 0xC8;
+    this->unk_1C8 = 0x3E80;
+    this->unk_1CA = 0x1F4;
+    this->unk_1CC = 0;
+    this->unk_1C0 = 0x10;
+}
+
+void ObjBean_Grown(ObjBean* this) {
+    this->unk_1C0--;
+    if (this->unk_1C0 == 6) {
+        this->unk_1CC = 0x7D0;
+    }
+    ObjBean_UpdateLeaves(this);
+    if (this->unk_1C2 <= 0) {
+        ObjBean_SetupLeavesStill(this);
+    }
+}
+
+void ObjBean_Init(Actor* thisx, GlobalContext* globalCtx) {
+    s32 path;
+    s32 linkAge;
+    ObjBean* this = THIS;
+
+    Actor_ProcessInitChain(&this->dyna.actor, sInitChain);
+    if (LINK_AGE_IN_YEARS == YEARS_ADULT) { // Flying bean
+        if (Flags_GetSwitch(globalCtx, this->dyna.actor.params & 0x3F) || (mREG(1) == 1)) {
+            path = (this->dyna.actor.params >> 8) & 0x1F;
+            if (path == 0x1F) {
+                osSyncPrintf(VT_COL(RED, WHITE));
+                // No path data?
+                osSyncPrintf("パスデータが無い？(%s %d)(arg_data %xH)\n", "../z_obj_bean.c", 909,
+                             this->dyna.actor.params);
+                osSyncPrintf(VT_RST);
+                Actor_Kill(&this->dyna.actor);
+                return;
+            }
+            if (globalCtx->setupPathList[path].count < 3) {
+                osSyncPrintf(VT_COL(RED, WHITE));
+                // Incorrect number of path data
+                osSyncPrintf("パスデータ数が不正(%s %d)(arg_data %xH)\n", "../z_obj_bean.c", 921,
+                             this->dyna.actor.params);
+                osSyncPrintf(VT_RST);
+                Actor_Kill(&this->dyna.actor);
+                return;
+            }
```

Comment:
else ifs are probably better.

Reply:
Since there are `Actor_Kill`s don't we usually just return?